TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs

```diff
@@ -424,15 +424,18 @@ public static IEnumerable<CompletionItem> GetCompletionForQualifiedName(Position
                     foreach (var variable in possibleVariables)
                     {
                         var children = new List<Node>();
-                        if (variable.Children != null && variable.Children.Count > 0) //It's a variable with levels inside
-                            children.AddRange(variable.Children);
-                        else //It's a typed variable, we have to search for children in the type
+                        if (variable.Children?.All(c => c is IndexDefinition) ?? true)
```

Comment:
It works, but I think the code is hard to understand.
Plus this code (like the previous one) need to call the SymbolTable.
We know have cached variable.
If it's a typed varialble you can use `variable.TypeDefinition` to get the TypeDefinition.

I propose this code:
```C#
                        var children = new List<Node>();

                        //if it's a typed variable, propose 1st children of the type
                        if (variable.TypeDefinition != null) 
                        {
                            children.AddRange(variable.TypeDefinition.Children.Where(t => t.Name != null || t.Children.Where(u => u.Name != null) != null));
                        }
                        
                        //It's a variable with levels inside
                        //Note that Index are also children
                        if (variable.Children?.Count > 0) 
                            children.AddRange(variable.Children);

                        //No children => nothing to do
```

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1605/commits/89a9149e4842f4faff4b936987bcaf014a59a81d